src/EtherFiNodesManager.sol

```diff
@@ -414,12 +419,33 @@ contract EtherFiNodesManager is
             // validate call
             if (data[i].length < 4) revert InvalidForwardedCall();
             bytes4 selector = bytes4(data[i][:4]);
-            if (!allowedForwardedEigenpodCalls[selector]) revert ForwardedCallNotAllowed();
+
+            // Check if user is allowed to call this selector on eigenpod
+            if (!allowedForwardedEigenpodCalls[msg.sender][selector]) revert ForwardedCallNotAllowed();
 
             returnData[i] = IEtherFiNode(nodes[i]).forwardEigenPodCall(data[i]);
         }
     }
 
+    /// @notice Batch update the whitelist for external calls (convenience function)
+    /// @param user The address to grant/revoke permissions for
+    /// @param selectors Array of method selectors
+    /// @param targets Array of call targets for forwarded calls
+    /// @param allowed Array of enable/disable flags
+    function batchUpdateAllowedForwardedExternalCalls(
```

Comment:
no need for batch endpoint. This method will be called very rarely and we are already low on space